schema.development.kf

```diff
@@ -426,6 +458,54 @@ action share_credential_through_dag (
     INSERT INTO shared_credentials (original_id, duplicate_id) VALUES ($original_credential_id, $id);
 }
 
+procedure share_credential_by_write_grant(
+    $id uuid,
+    $human_id uuid,
+    $original_credential_id uuid,
+    $credential_type text,
+    $credential_level text,
+    $credential_status text,
+    $content text,
+    $encryption_public_key text,
+    $grantee text,
+    $locked_until int,
+    $issuer text) public {
+
+    if $locked_until < 0 {
+        error('locked_until must be positive integer timestamp or zero');
+    }
+
+    $wg_id := get_write_grant_id($human_id); // throw an error if no write_grant
+
+    if !credential_belongs_to_human($original_credential_id, $human_id) {
+        error('the original credential does not belong to the human');
+    }
+
+    if $wg_id::TEXT != get_credential_inserter($original_credential_id) {
+        error('you can share only original credentials you created');
+    }
+
+    $generated_uuid := uuid_generate_v5('31276fd4-105f-4ff7-9f64-644942c14b79'::uuid, format('%s-%s-%s', $grantee, $original_credential_id, $locked_until));
```

Comment:
Maybe we want to re-use the same logic that's in the contracts? See https://github.com/idos-network/idos-access-grants/blob/7431ee59feb6ddf891888402b3673f1fe18df4b6/near-rs/contract/src/lib.rs#L53

I'm 99% inclined to say "no, nobody cares about this id". Just wanted to raise awareness that this will make internal and external grants even more different.